nixos/modules/services/x11/redshift.nix

```diff
@@ -88,51 +69,90 @@ in {
       example = "/bin/redshift-gtk";
```

Comment:
Since we are already breaking the backwards compatibility with this module I am thinking of changing this option to:

```nix
{
  type = types.path;
  default = ${pkgs.redshift}/bin/redshift;
}
```

This way, we at least have some type checking in here, but on the other hand this may make this module more error prone since the user may forget to set the correct `cfg.package` option (that is still necessary for https://github.com/NixOS/nixpkgs/pull/108739/files#diff-61e7d28082c9dd1fe9bd0282f55b1cd33f1397fa87b424b6fdf71253e591bc02R125).

If someone has a better solution, suggestions are welcome too.

Reply:
Add an assertion that checks if the executable actually exists. This should fix most issues.